mpas_analysis/shared/timekeeping/MpasRelativeDelta.py

```diff
@@ -0,0 +1,131 @@
+import datetime
+from dateutil.relativedelta import relativedelta
+import calendar
+
+
+class MpasRelativeDelta(relativedelta):
+    """
+    MpasRelativeDelta is a subclass of dateutil.relativedelta for relative time
+    intervals with different MPAS calendars.
+
+    Only relative intervals (years, months, etc.) are supported and not the
+    absolute date specifications (year, month, etc.).  Addition/subtraction
+    of datetime.datetime objects (but not other MpasRelativeDelta,
+    datetime.timedelta or other related objects) is supported.
```

Comment:
We should test all overloaded operator operations too because this is best practices and will ensure that we get what is expected.